qa/wallet.sh

```diff
@@ -115,132 +116,130 @@ tpc0=$(echo ${test0} | jq ".peer_count")
+		exit 1
+	else
+		echo "--> Got ${tc} ✓"
+	fi
+
+	echo ""
+	echo "--> Checking peer${i}'s files count..."
+	if [[ "$fc" -ne "$fc0" ]]; then
+		echo ERROR: peer ${i} has incorrect "files_count": ${fc}. Expected ${fc0}.
+		exit 1
+	else
+		echo "--> Got ${fc} ✓"
+	fi
+
+	echo ""
+	echo "--> Checking peer${i}'s contact count..."
+	if [[ "$cc" -ne "$cc0" ]]; then
+		echo ERROR: peer ${i} has incorrect "contact_count": ${cc}. Expected ${cc0}.
+		exit 1
+	else
+		echo "--> Got ${cc} ✓"
+	fi
+
+	# account thread
+	account=$(textile threads get $(textile config Account.Thread))
+	ath=$(echo ${account} | jq ".head")
+	atbc=$(echo ${account} | jq ".block_count")
+	atpc=$(echo ${account} | jq ".peer_count")
+
+	echo ""
+	echo "--> Checking peer${i}'s account thread HEAD..."
+	if [[ "$ath" != "$ath0" ]]; then
+		echo ERROR: peer ${i} has incorrect "head": ${ath}. Expected ${ath0}.
+		exit 1
+	else
+		echo "--> Got ${ath} ✓"
+	fi
+
+	echo ""
+	echo "--> Checking peer${i}'s account thread block count..."
+	if [[ "$atbc" -ne "$atbc0" ]]; then
+		echo ERROR: peer ${i} has incorrect "block_count": ${atbc}. Expected ${atbc0}.
+		exit 1
+	else
+		echo "--> Got ${atbc} ✓"
+	fi
+
+	echo ""
+	echo "--> Checking peer${i}'s account thread peer count..."
+	if [[ "$atpc" -ne "$atpc0" ]]; then
+		echo ERROR: peer ${i} has incorrect "peer_count": ${atpc}. Expected ${atpc0}.
+		exit 1
+	else
+		echo "--> Got ${atpc} ✓"
+	fi
+
+	# test thread
+	test=$(textile threads get ${thread})
+	th=$(echo ${test} | jq ".head")
+	tbc=$(echo ${test} | jq ".block_count")
+	tpc=$(echo ${test} | jq ".peer_count")
+
+	echo ""
+	echo "--> Checking peer${i}'s test thread HEAD..."
+	if [[ "$th" != "$th0" ]]; then
+		echo ERROR: peer ${i} has incorrect "head": ${th}. Expected ${th0}.
+		exit 1
+	else
+		echo "--> Got ${th} ✓"
+	fi
+
+	echo ""
+	echo "--> Checking peer${i}'s test thread block count..."
+	if [[ "$tbc" -ne "$tbc0" ]]; then
+		echo ERROR: peer ${i} has incorrect "block_count": ${tbc}. Expected ${tbc0}.
+		exit 1
+	else
+		echo "--> Got ${tbc} ✓"
+	fi
+
+	echo ""
+	echo "--> Checking peer${i}'s test thread peer count..."
+	if [[ "$tpc" -ne "$tpc0" ]]; then
+		echo ERROR: peer ${i} has incorrect "peer_count": ${tpc}. Expected ${tpc0}.
+		exit 1
+	else
+		echo "--> Got ${tpc} ✓"
+	fi
 done
 
 # draw thread graphs
-for i in `seq 1 $((num - 1))`;
-do
-    export API=$(getApi i)
-    textile blocks ls -t $(textile config Account.Thread) -l 100 -d | dot -Tpng -o /tmp/peer${i}_account_blocks.png
-    textile blocks ls -t ${thread} -l 100 -d | dot -Tpng -o /tmp/peer${i}_thread_blocks.png
+for i in $(seq 1 $((num - 1))); do
+	export API=$(getApi i)
+	textile blocks ls -t $(textile config Account.Thread) -l 100 -d | dot -Tpng -o /tmp/peer${i}_account_blocks.png
+	textile blocks ls -t ${thread} -l 100 -d | dot -Tpng -o /tmp/peer${i}_thread_blocks.png
 done
 
 echo "Success."
 
-if [[ "$wait" = "yes" ]]; then
-    wait
+if [[ "$wait" == "yes" ]]; then
```

Comment:
Seems like it caught a bug here.

Reply:
https://stackoverflow.com/a/2237103/641834

Reply:
New conventions? It was working fine... maybe not in newer shells.

Reply:
I think we may need to adopt https://github.com/textileio/meta/issues/2 as the different shell scripts in this repo are using different shells.

This particular file is using bash, despite the `.sh` extension.

Bash treats `=` and `==` the same within square brackets, so the formatter swapped it for the option that reduces user error I guess.

So this is a non-functional change as expected.

We should take steps though to ensure our shell scripts are consistent, ref:

- https://github.com/textileio/meta/issues/5
- https://github.com/textileio/meta/issues/2
- https://github.com/textileio/meta/issues/1

Reply:
Totally. Wow, I'm sold on the formatter.